lib/config.js

```diff
@@ -9,6 +9,7 @@ config['messages-client'] = config['messages-client'] || {};
 config['messages-client'].clients = config['messages-client'].clients || [];
 config['messages-client'].maxMessageRetrievalRetries =
   config['messages-client'].maxMessageRetrievalRetries || 3;
+config['messages-client'].stopOnMaxRetries = true;
```

Comment:
Here are some suggested names:

Error name when you can't connect/bad request/whatever: `MessagePollError`. Include `httpStatusCode` in the details.

To configure how many failed poll errors to tolerate before notifying someone: `maxPollErrorsBeforeNotify`.

To configure if polling should cease after maximum is reached: `stopPollingAfterNotify`.